pygmt/tests/test_helpers.py

```diff
@@ -113,3 +116,33 @@ def test_args_in_kwargs():
     # Failing list of arguments
     failing_args = ["D", "E", "F"]
     assert not args_in_kwargs(args=failing_args, kwargs=kwargs)
+
+
+def test_tempfile_from_geojson():
+    """
+    Test tempfile_from_geojson works when passed a geopandas GeoDataFrame.
+    """
+    linestring = shapely.geometry.LineString([(20, 15), (30, 15)])
+    polygon = shapely.geometry.Polygon([(20, 10), (23, 10), (23, 14), (20, 14)])
+    gdf = gpd.GeoDataFrame(
+        index=["polygon", "linestring"],
+        geometry=[polygon, linestring],
+    )
+    with tempfile_from_geojson(gdf) as geojson_string:
+        assert isinstance(geojson_string, str)
+        with open(geojson_string) as tmpfile:
+            assert os.path.basename(tmpfile.name).startswith("pygmt-")
+            assert os.path.basename(tmpfile.name).endswith(".gmt")
```

Comment:
> This is failing tests on the Python 3.7 CI jobs with `ModuleNotFoundError: No module named 'geopandas'`. Should `geopandas` be added/not made optional to requirements.txt and environment.yml?

No, `geopandas` should be an optional-only dependency. Could you provide some context as to why these two tests for the tempfile_from_geojson() function are to be added? To be honest, this function (added in #1000) is meant to be temporary until https://github.com/GenericMappingTools/gmt/issues/4599 is resolved (though that might take some time).

If you must though, I'd suggest using [pytest.importorskip](https://docs.pytest.org/en/4.6.x/reference.html#pytest-importorskip) to load geopandas and shapely, similar to at https://github.com/GenericMappingTools/pygmt/blob/b2808cf30e5e198515deabf17368ef0731bc84ab/pygmt/tests/test_geopandas.py#L8-L9

Reply:
The reason I added it was that the code coverage was showing `tempfile_from_geojson()` wasn't tested. I just assumed we hadn't gotten around to writing tests for it yet so I wrote them.

Reply:
The tempfile_from_geojson function does have full test coverage (see https://codecov.io/gh/GenericMappingTools/pygmt/src/master/pygmt/helpers/tempfile.py). It is tested indirectly at test_geopandas.py.

Reply:
Sounds good, I'll close the PR.